reference/src/taimpl/src/internal/ta.c

```diff
@@ -371,6 +372,144 @@ static sa_status ta_invoke_key_import(
             parameters, context->client, uuid);
 }
 
+static sa_status ta_invoke_key_provision(
+        sa_key_import_s* key_import,
+        const uint32_t param_types[NUM_TA_PARAMS],
+        ta_param params[NUM_TA_PARAMS],
+        const ta_session_context* context,
+        const sa_uuid* uuid) {
+
+    if (CHECK_NOT_TA_PARAM_INOUT(param_types[0]) || params[0].mem_ref_size != sizeof(sa_key_import_s) ||
+        CHECK_NOT_TA_PARAM_IN(param_types[1]) ||
+        (CHECK_NOT_TA_PARAM_IN(param_types[2]) && CHECK_NOT_TA_PARAM_NULL(param_types[2], params[2])) ||
+        (CHECK_NOT_TA_PARAM_IN(param_types[3]) && CHECK_NOT_TA_PARAM_NULL(param_types[3], params[3]))) {
+        ERROR("Invalid param[0] or param type");
+        return SA_STATUS_INVALID_PARAMETER;
+    }
+
+    if (params[1].mem_ref == NULL || params[1].mem_ref_size == 0) {
+        ERROR("NULL param[1]");
+        return SA_STATUS_NULL_PARAMETER;
+    }
+
+    if (key_import->api_version != API_VERSION) {
+        ERROR("Invalid api_version");
+        return SA_STATUS_INVALID_PARAMETER;
+    }
+
+    sa_import_parameters_soc parameters_soc;
+    void* parameters = NULL;
+    // params[2].mem_ref can be null.
```

Comment:
this comment is a bit unclear.  the comment says, params[2].mem_ref can be null, but the following lines return an error if it is null.